### PR TITLE
Comment on all PR failures by default and add gubernator link

### PR DESCRIPTION
**Pull request by fejta**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/646)

### CODE REVIEW

File: ciongke/cmd/hook/main.go [hunk 1]
```diff
@@ -63,7 +63,6 @@ var defaultJenkinsJobs = map[string][]JenkinsJob{
 			Trigger:          regexp.MustCompile(`@k8s-bot (gce )?(e2e )?test this`),
 			AlwaysRun:        true,
 			Context:          "Jenkins GCE e2e",
-			CommentOnFailure: true,
```
Comment (spxtr) [inline ciongke/cmd/hook/main.go]: The default for bool is false in Go, and we explicitly pass this to test-pr.

File: ciongke/cmd/test-pr/main.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ const (
 
 	// Inputs are context, URL, commit, retest command.
 	// The deletion logic requires that it start with context.
-	bodyFormat = `%s [**failed**](%s) for commit %s.
+	bodyFormat = `%s [**failed**](%s) for commit %s (see full [pr test history](http://ci-test.k8s.io/%s)).
```
Comment (spxtr) [inline ciongke/cmd/test-pr/main.go]: `%s [**failed**](%s) for commit %s. [Full PR test history](http://ci-test.k8s.io/%s).`

Also, update the comment with the new input.

Comment (fejta) [inline ciongke/cmd/test-pr/main.go]: Updated by eliminating constant literals which are only used once.

File: ciongke/cmd/hook/main.go (hunk 1 above)
Comment (fejta) [inline ciongke/cmd/hook/main.go]: Done

File: ciongke/cmd/hook/main.go [hunk 3]
```diff
@@ -51,67 +51,77 @@ var defaultJenkinsJobs = map[string][]JenkinsJob{
...
 		{
 			Name:    "fejta-pull-unit",
 			Trigger: regexp.MustCompile(`@fejta do a little dance`),
 			Context: "Jenkins fejta unit",
+			RerunCommand:     "@fejta do a little dance",
 		},
 		{
 			Name:    "fejta-pull-e2e",
-			Trigger: regexp.MustCompile(`@fejta do a little dance`),
+			Trigger: regexp.MustCompile(`@fejta make a little love`),
```
Comment (rmmh) [inline ciongke/cmd/hook/main.go]: uhhh

**Comment by ixdy** [comment]
So if a test fails 6 builds, this will cause 6 comments? I'm not sure we want that.

**Comment by fejta** [comment]
Is this everything we want? No.

Does this tell people how to retry any PR test that fails? Yes.

Is that better than the current situation where we leave people confused and in the dark? Definitely.

File: ciongke/cmd/hook/main.go (hunk 3 above)
Comment (fejta) [inline ciongke/cmd/hook/main.go]: It is the next line in the song! Want me to change it?

**Comment by spxtr** [comment]
It's not ideal, for sure. I'd like to make it delete and recreate an aggregate comment, but there's some issues with doing that naively.

In the meantime, I get a lot of questions from people saying "What's the command to rerun such and such?" which this should cut back on.

File: ciongke/cmd/hook/githubagent.go [hunk 4]
```diff
@@ -52,9 +52,9 @@ type JenkinsJob struct {
 	// eg @k8s-bot e2e test this
 	Trigger *regexp.Regexp
 	// Whether or not to comment on GitHub on failure.
-	CommentOnFailure bool
+	SkipFailureComment bool
```
Comment (spxtr) [inline ciongke/cmd/hook/githubagent.go]: Might as well delete SkipFailureComment as well as CommentOnFailure in test-pr. I can take care of that in a follow-up PR.

File: ciongke/cmd/test-pr/main.go (hunk 2 above)
Comment (spxtr) [inline ciongke/cmd/test-pr/main.go]: That looks good. I'd still like the comment text to have my change.

File: ciongke/cmd/hook/githubagent.go (hunk 4 above)
Comment (fejta) [inline ciongke/cmd/hook/githubagent.go]: Done, although keeping the flag for now and also sending the flag until the next push per standard flag deprecation practice.

File: ciongke/cmd/hook/main.go (hunk 3 above)
Comment (fejta) [inline ciongke/cmd/hook/main.go]: Done

File: ciongke/cmd/test-pr/main.go (hunk 2 above)
Comment (fejta) [inline ciongke/cmd/test-pr/main.go]: Done

File: ciongke/cmd/hook/githubagent.go (hunk 4 above)
Comment (spxtr) [inline ciongke/cmd/hook/githubagent.go]: Ok.

File: ciongke/cmd/hook/githubagent.go [hunk 5]
```diff
@@ -52,9 +52,6 @@ type JenkinsJob struct {
 	// eg @k8s-bot e2e test this
 	Trigger *regexp.Regexp
 	// Whether or not to comment on GitHub on failure.
-	CommentOnFailure bool
-	// What should users use to retest just this job? This needs to match
```
Comment (spxtr) [inline ciongke/cmd/hook/githubagent.go]: This comment should be kept, delete the CommentOnFailure comment.

**Comment by spxtr** [comment]
Run gofmt and address comment then this looks good.

File: ciongke/cmd/hook/githubagent.go (hunk 5 above)
Comment (fejta) [inline ciongke/cmd/hook/githubagent.go]: Done

File: ciongke/cmd/test-pr/main.go [hunk 6]
```diff
@@ -229,7 +215,11 @@ func (c *testClient) tryCreateFailureComment(url string) {
 			}
 		}
 	}
-	body := fmt.Sprintf(bodyFormat, c.Context, url, c.Commit, c.RerunCommand)
+	// The deletion logic requires that it start with context.
+	bodyFormat := `%s [**failed**](%s) for commit %s. [Full PR test history](http://ci-test.k8s.io/%s).
+
+The magic incantation to run this job again is ` + "`%s`" + `. Please help us cut down flakes by linking to an [open flake issue](https://github.com/kubernetes/kubernetes/issues?q=is:issue+label:kind/flake+is:open) when you hit one in your PR.`
+	body := fmt.Sprintf(bodyFormat, c.Context, c.PRNumber, url, c.Commit, c.RerunCommand)
```
Comment (spxtr) [inline ciongke/cmd/test-pr/main.go]: Are PRNumber and url backwards?

Comment (fejta) [inline ciongke/cmd/test-pr/main.go]: PRNumber is supposed to go after Commit, who moved it?!

**Pull request merged**